sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -814,4 +816,50 @@ object DDLUtils {
       }
     }
   }
+
+  /**
+   * ALTER TABLE ADD COLUMNS command does not support temporary view/table,
+   * view, or datasource table with text, orc formats or external provider.
+   */
+  def verifyAlterTableAddColumn(
+      catalog: SessionCatalog,
+      table: TableIdentifier): CatalogTable = {
+    if (catalog.isTemporaryTable(table)) {
+      throw new AnalysisException(
+        s"${table.toString} is a temporary VIEW, which does not support ALTER ADD COLUMNS.")
+    }
+
+    val catalogTable = catalog.getTableMetadata(table)
+    if (catalogTable.tableType == CatalogTableType.VIEW) {
+      throw new AnalysisException(
+        s"${table.toString} is a VIEW, which does not support ALTER ADD COLUMNS.")
+    }
+
+    if (isDatasourceTable(catalogTable)) {
+      catalogTable.provider.get match {
+        case provider if provider.toLowerCase == "text" =>
+          // TextFileFormat can not support adding column either because text datasource table
+          // is resolved as a single-column table only.
+          throw new AnalysisException(
+            s"""${table.toString} is a text format datasource table,
+               |which does not support ALTER ADD COLUMNS.""".stripMargin)
+        case provider if provider.toLowerCase == "orc"
+          || provider.startsWith("org.apache.spark.sql.hive.orc") =>
```

Comment:
When we store the metadata in the catalog, we unify different representations to `orc`, right? Can you find any case to break it?